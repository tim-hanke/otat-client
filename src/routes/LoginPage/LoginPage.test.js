import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LoginPage from "./LoginPage";

describe("LoginPage component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <LoginPage />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LoginPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
