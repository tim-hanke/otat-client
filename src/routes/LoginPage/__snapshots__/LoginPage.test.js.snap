// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders the UI as expected 1`] = `
<section
  className="Section LoginPage"
>
  <h2>
    Login
  </h2>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <p>
      Demo account: demo@test.com/Pass123!
    </p>
    <div
      className="email"
    >
      <label
        htmlFor="LoginForm__email"
      >
        Email
      </label>
      <input
        className="Input "
        id="LoginForm__email"
        name="email"
        placeholder="demo@test.com"
        required={true}
        type="email"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <input
        className="Input "
        id="LoginForm__password"
        name="password"
        placeholder="Pass123!"
        required={true}
        type="password"
      />
    </div>
    <button
      className="Button "
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
